etc/scripts/freeze_and_update_reqs.py

```diff
@@ -0,0 +1,134 @@
+from shutil import copy, rmtree
+import sys
+
+python_version = str(sys.version_info[0]) + str(sys.version_info[1])
+py_abi = "{0}cp{1}{0}".format("*", python_version)
+
+
+def generate_req_text(input_dir, req_file, package_name, upgrade):
+    """
+    Generate a requirement file at `output_file`(by default requirements.txt) of all dependencies wheels and sdists present in the `input_dir` directory.
+    If a `package_name` is provided it will be updated to its latest version.
+    """
+    thirdparty = resource_iter(input_dir, with_dirs=False)
+    dependencies = [
+        files
+        for files in thirdparty
+        if fnmatch.fnmatchcase(files, '*py3*')
+        or fnmatch.fnmatchcase(files, py_abi)
+        or (
+            fnmatch.fnmatchcase(files, '*tar.gz*')
+            and not fnmatch.fnmatchcase(files, '*py2-ipaddress-3.4.1.tar.gz*')
+        )
+    ]
+    if not (os.path.isdir('temp dir')):
+        os.mkdir('temp dir')
+    for deps in dependencies:
+        copy(deps, 'temp dir')
+    pip_args = [
+        'pip-compile',
+        '--generate-hashes',
+        '--find-links',
+        'temp dir',
+        '--output-file',
+        req_file,
+        '--allow-unsafe',
+        '--pip-args',
+        '--no-index',
+    ]
+    if upgrade:
+        pip_args.append('--upgrade')
+    if package_name:
+        pip_args.extend(['--upgrade-package', package_name])
+    run(pip_args)
+    rmtree('temp dir')
+
+
+def main_with_args(args: str) -> None:
+    parser = argparse.ArgumentParser(
+        description="""Generate a requirement file at `output_file`(by default requirements.txt) of all dependencies wheels and sdists present in the `input_dir` directory.
+    If a `package_name` is provided it will be updated to its latest version.
+EXAMPLE:
+freeze_and_update_reqs.py \\
+  --deps_directory DEPS_DIRECTORY \\
+  --output OUTPUT \\
+  --upgrade_package PACKAGE_NAME \\
+""",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        '--deps_directory',
+        help='Required: Thirdparty Dependencies directory to be archived.',
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        '--requirement',
+        help='Requirement file name. Required if more than one input file is given. Will be derived from input file otherwise.',
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        '--upgrade',
+        help='Upgrade all dependencies to new version.',
+        action='store_true',
+    )
+
+    parser.add_argument(
+        '--upgrade_package',
+        help='Specify particular packages to upgrade.',
+        type=str,
+        default=None,
+    )
+
+    args = parser.parse_args()
+
+    tpdir = args.deps_directory
+    requirement_file = args.requirement
+    package_name = args.upgrade_package
+    upgrade = args.upgrade or False
+    generate_req_text(tpdir, requirement_file, package_name, upgrade)
+
```

Comment:
prefer named keyword arguments for example:
`(tpdir=tpdir, requirement_file=requirement_file, package_name=package_name, upgrade=upgrade)
But do not use these names... use the same variable names on both sides:  e.g. if you use `args.deps_directory`  then use `deps_directory` and not `tpdir`